src/plugins/discover/public/context_awareness/profile_providers/observability/observability_root_profile/accessors/get_app_menu.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { AppMenuActionId, AppMenuActionType, AppMenuRegistry } from '@kbn/discover-utils';
+import { DATA_QUALITY_LOCATOR_ID, DataQualityLocatorParams } from '@kbn/deeplinks-observability';
+import { OBSERVABILITY_THRESHOLD_RULE_TYPE_ID } from '@kbn/rule-data-utils';
+import { isOfQueryType } from '@kbn/es-query';
+import { AppMenuExtensionParams } from '../../../..';
+import type { RootProfileProvider } from '../../../../profiles';
+import { ProfileProviderServices } from '../../../profile_provider_services';
+
+export const createGetAppMenu =
+  (services: ProfileProviderServices): RootProfileProvider['profile']['getAppMenu'] =>
+  (prev) =>
+  (params) => {
+    const prevValue = prev(params);
+
+    return {
+      appMenuRegistry: (registry) => {
+        // Register custom link actions
+        registerDatasetQualityLink(registry, services);
+        // Register alerts sub menu actions
+        registerCreateSLOAction(registry, services, params);
+        registerCustomThresholdRuleAction(registry, services, params);
+
+        return prevValue.appMenuRegistry(registry);
+      },
+    };
+  };
+
+const registerDatasetQualityLink = (
+  registry: AppMenuRegistry,
+  { share, timefilter }: ProfileProviderServices
+) => {
+  const dataQualityLocator =
+    share?.url.locators.get<DataQualityLocatorParams>(DATA_QUALITY_LOCATOR_ID);
+
+  if (dataQualityLocator) {
+    registry.registerCustomAction({
+      id: 'dataset-quality-link',
+      type: AppMenuActionType.custom,
+      controlProps: {
+        label: 'Data sets',
```

Comment:
Can we use i18n for this?

Reply:
Addressed in [refactor(discover): use i18n](https://github.com/elastic/kibana/pull/198987/commits/8c073e23cff9d1566718a93645c8a522cf2e9980)